src/api/app/services/trigger_controller_service/scm_extractor.rb

```diff
@@ -10,187 +16,10 @@ def initialize(scm, event, payload)
-                       # We need this for SCMStatusReporter#call
-                       target_repository_full_name: @payload.dig(:repository, :full_name),
-                       ref: payload_ref,
-                       # We need this for Workflow::Step#branch_request_content_{github,gitlab}
-                       commit_sha: @payload[:after],
-                       # We need this for Workflows::YAMLDownloader#download_url
-                       # when the push event is for commits, we get the branch name from ref.
-                       target_branch: payload_ref.sub('refs/heads/', '')
-                     })
-
-      return unless payload_ref.start_with?('refs/tags/')
-
-      # We need this for Workflow::Step#target_package_name
-      # 'target_branch' will contain a commit SHA
-      payload.merge!({ tag_name: payload_ref.sub('refs/tags/', ''),
-                       target_branch: @payload.dig(:head_commit, :id),
-                       commit_sha: @payload.dig(:head_commit, :id) })
-    end
-
-    def gitlab_payload_tag(payload)
-      payload.merge!({ # We need this for Workflow::Step#target_package_name
-                       tag_name: @payload[:ref].sub('refs/tags/', ''),
-                       # We need this for Workflows::YAMLDownloader#download_url
-                       # This will contain a commit SHA
-                       target_branch: @payload[:after],
-                       # We need this for Workflows::YAMLDownloader#download_url
-                       path_with_namespace: @payload.dig(:project, :path_with_namespace),
-                       # We need this for SCMWebhookEventValidator#valid_push_event
-                       ref: @payload[:ref],
-                       # We need this for Workflow::Step#branch_request_content_{github,gitlab}
-                       commit_sha: @payload[:after]
-                     })
-    end
-
-    def gitea_payload_push(payload)
-      payload_ref = @payload.fetch(:ref, '')
-      payload.merge!({
-                       # We need this for Workflow::Step#branch_request_content_github
-                       source_repository_full_name: @payload.dig(:repository, :full_name),
-                       # We need this for SCMStatusReporter#call
-                       target_repository_full_name: @payload.dig(:repository, :full_name),
-                       ref: payload_ref,
-                       # We need this for Workflow::Step#branch_request_content_{github,gitlab}
-                       commit_sha: @payload[:after],
-                       # We need this for Workflows::YAMLDownloader#download_url
-                       # when the push event is for commits, we get the branch name from ref.
-                       target_branch: payload_ref.sub('refs/heads/', '')
-                     })
-
-      return unless payload_ref.start_with?('refs/tags/')
+      extractor = SCM_EXTRACTORS[@scm]&.new(@event, @payload)
```

Comment:
The instantiation could happen below after the `return ...` line. So using the `&` operator would not be needed.